src/prefect/tasks/cubejs/cubejs_utils.py

```diff
@@ -0,0 +1,125 @@
+import jwt
+from requests import Session
+import time
+from typing import Dict, Union
+
+from prefect.engine.signals import FAIL
+
+# Cube Cloud base URL
+__CUBEJS_CLOUD_BASE_URL = "https://{subdomain}.cubecloud.dev"
+
+
+def get_cube_base_url(subdomain: str, url: str) -> str:
```

Comment:
I like that you refactored the logic for calling the cube.js API into separate utils. I would recommend making a `CubejsClient` class to improve encapsulation. 

With a a `CubejsClient` class you could initialize the client with a `url`/`subdomain`, `security_context`, and `secret`. Then you can have `query` and `get_generated_sql` as your public methods. API token and URL construction then become implementation details. The Databricks tasks are an example of this pattern.